test/Infrastructure.IntegrationTest/AdminConsole/Migrations/FinalFlexibleCollectionsDataMigrationsTests.cs

```diff
@@ -0,0 +1,156 @@
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user was migrated to a User type with null permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.User, migratedOrgUser.Type);
+        Assert.Null(migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = "FinalFlexibleCollectionsDataMigrations")]
+    public async Task RunMigration_WithDeleteAssignedCollections_WithCustomUserType_MigratesToUserNullPermissions(
+        IUserRepository userRepository,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IMigrationTesterService migrationTester)
+    {
+        // Setup data
+        var orgUser = await SetupData(
+            userRepository, organizationRepository, organizationUserRepository,
+            editAssignedCollections: false, deleteAssignedCollections: true);
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user was migrated to a User type with null permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.User, migratedOrgUser.Type);
+        Assert.Null(migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = "FinalFlexibleCollectionsDataMigrations")]
+    public async Task RunMigration_WithEditAndDeleteAssignedCollections_WithCustomUserType_MigratesToUserNullPermissions(
+        IUserRepository userRepository,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IMigrationTesterService migrationTester)
+    {
+        // Setup data
+        var orgUser = await SetupData(
+            userRepository, organizationRepository, organizationUserRepository,
+            editAssignedCollections: true, deleteAssignedCollections: true);
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user was migrated to a User type with null permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.User, migratedOrgUser.Type);
+        Assert.Null(migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = "FinalFlexibleCollectionsDataMigrations")]
+    public async Task RunMigration_WithoutAssignedCollectionsPermissions_WithCustomUserType_RemovesAssignedCollectionsPermissions(
```

Comment:
Would be good to set some other permissions to `true` and then make sure their values are preserved.

Reply:
Added `accessEventLogs = true`